### PR TITLE
Argument checks

### PR DESCRIPTION
**Pull request by maximerety**

### CODE REVIEW

File: spec/tanker/user_token_spec.rb [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
+      }.to raise_exception(TypeError)
+    end
+  end
+
+  it 'raises an ArgumentError if argument is an invalid base64 string' do
+    args = [@trustchain[:id], @trustchain[:sk], @user_id]
+    invalid_base64 = ['&:,?', '&:,?']
 
-    user_id = 'matz@tanker.io'
+    2.times do |position|
```
Comment (dmerejkowsky) [inline spec/tanker/user_token_spec.rb]: ditto

File: spec/tanker/user_token_spec.rb [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
+
+  it 'has a version number' do
+    expect(Tanker::UserToken::VERSION).not_to be nil
+  end
+
+  it 'raises a TypeError if argument is not a string' do
+    args = [@trustchain[:id], @trustchain[:sk], @user_id]
+    not_a_string = [nil, {}, 1234]
+
+    3.times do |position|
```
Comment (dmerejkowsky) [inline spec/tanker/user_token_spec.rb]: This is really hard to read. Why not something like `not_at_string.each`  ?

**Review by dmerejkowsky — CHANGES_REQUESTED** [review]

File: spec/tanker/user_token_spec.rb (hunk 2 above)
Comment (maximerety) [reply spec/tanker/user_token_spec.rb]: Changed in a regular `for` loop. Not so common in Ruby but makes things clearer here.

File: spec/tanker/user_token_spec.rb (hunk 1 above)
Comment (maximerety) [reply spec/tanker/user_token_spec.rb]: ditto

**Review by dmerejkowsky — APPROVED** [review]

**Pull request merged**